cmd/clefui/main.go

```diff
@@ -15,20 +15,21 @@ func main() {
 	// Make all the channels
 	uiClose := make(chan bool)
 	appCancel := make(chan os.Signal, 1)
-	readyToClose := make(chan bool)
 	readyToStart := make(chan string)
 
 	// notify appCancel channel on Ctrl + C
 	signal.Notify(appCancel, os.Interrupt)
 
 	// trap Ctrl+C and call cancel on the context
 	ctx, cancel := context.WithCancel(context.Background())
+	_ = cancel
```

Comment:
?
This is a no-op. 
In general, I'm not sure what you're trying to do here. The previous code looked for a cancel, and then invoked the `cancel` method which could then traverse along everyone who use the same context.

Reply:
I might have been a bit too aggressive with the cleanup here... My understanding is that this listens to OS signals and kills the app if relevant (at least the previous handler was looking like this). Which doesn't make sense when relying on Qt as it's handling that in the event loop already. So either we should do some cleanup based on a Qt signal, or remove this call. Then this line is just there because that's the way I found to make it compile in go, since an unused variable is an error. I tested with and without the call to the cancel function and didn't notice any difference...

Reply:
You can leave it as is. Althoguh it would be more proper to not declare `cancel`, just `ctx, _ := context.WithCancel(context.Background())`

Reply:
Except, then it doesn't make sense to use `WithCancel`. Just use `ctx := context.BackGround()`

Reply:
Indeed, I didn't know the API allowed that. I'll fix that!

Reply:
... And then you run into the problem: 
```
external.NewServer(ctx, stdin, stdout)
```
How do you tell the server to stop without a cancel? 
I guess the main thread just dies, and that's that.... Myeah, if you leave it as is, then at least it's obvious that something's fishy, and I can fix it later

Reply:
I read a bit more about this cancel function, I thought it was at OS signal level, but it seems to be golang specific and related to http. So yeah, then it makes sense to call it (surprised go doesn't scream when you don't quit a process properly though). I'll get back to the previous code ;)